target_redshift/db_sync.py

```diff
@@ -374,27 +378,36 @@ def load_csv(self, s3_key, count):
                 cur.execute(self.drop_table_query(is_stage=True))
                 cur.execute(self.create_table_query(is_stage=True))
 
-                # Step 2: Generate copy options - Override defaults from config.json if defined
+                # Step 2: Generate copy credentials - prefer role if provided, otherwise use access and secret keys
+                copy_credentials = """
+                    iam_role '{aws_role_arn}'
+                """.format(aws_role_arn=self.connection_config['aws_redshift_copy_role_arn']) if "aws_redshift_copy_role_arn" in self.connection_config else """
+                    ACCESS_KEY_ID '{aws_access_key_id}'
```

Comment:
Seems like there is a little bug here, and some cases it's causing redshift to hang on the COPY command, locking tables and not responding for long minutes.

The reason is that in `utils.py` you set `config['aws_redshift_copy_role_arn'] = None` and that producing this COPY:
```
COPY .... .csv

                    iam_role 'None'

                    EMPTYASNULL BLANKSASNULL TRIMBLANKS TRUNCATECOLUMNS
                    TIMEFORMAT 'auto'
                    COMPUPDATE OFF STATUPDATE OFF

                    DELIMITER ',' REMOVEQUOTES ESCAPE
```

Please note the `iam_role 'None'`! This puts a lock onto the table and COPY never finishes for some reason.

Replacing `if "aws_redshift_copy_role_arn" in self.connection_config` to if `self.connection_config.get("aws_redshift_copy_role_arn")` solves the problem. Can you please change it and use .get avoiding passing None?

Reply:
Done, thanks for the suggestion.